### PR TITLE
feat(datadog): add tracer

### PR DESCRIPTION
**Pull request by sileht**
Change-Id: Id3de09ed78020e143126f72f09e9c7e020593348

### CODE REVIEW

**Review by jd — CHANGES_REQUESTED** [review]
requirements?

**Review by jd — COMMENTED** [review]
also what about onprem?

File: mergify_engine/logs.py [hunk 1]
```diff
@@ -70,6 +71,17 @@ def add_fields(self, log_record, record, message_dict):
         if worker_id is not None:
             log_record.update({"worker_id": worker_id})
 
+        span = ddtrace.tracer.current_span()
```
Comment (jd) [inline mergify_engine/logs.py:74]: https://github.com/DataDog/dd-trace-py/blob/master/ddtrace/tracer.py#L261

File: Dockerfile [hunk 2]
```diff
@@ -36,5 +36,6 @@ WORKDIR /app
 ENV VIRTUAL_ENV=/venv
 ENV PYTHONUNBUFFERED=1
 ENV DD_DOGSTATSD_DISABLE=1
+ENV DD_APM_ENABLED=0
```
Comment (jd) [inline Dockerfile:39]: This controls the Datadog agent, not the tracer library.

File: mergify_engine/web/asgi.py [hunk 3]
```diff
@@ -22,4 +22,5 @@
 if config.SENTRY_URL:
     application = SentryAsgiMiddleware(application)  # type:ignore[assignment]
 
+tracer.configure(service="web")
```
Comment (jd) [inline mergify_engine/web/asgi.py]: I fear this is going to be conflicting soon with the dashboard.
Rather use `engine-web`.

File: mergify_engine/worker.py [hunk 4]
```diff
@@ -1036,6 +1041,7 @@ async def run_forever() -> None:
 
 
 def main() -> None:
+    tracer.configure(service="worker")
```
Comment (jd) [inline mergify_engine/worker.py]:
```suggestion
    tracer.configure(service="engine-worker")
```

**Review by jd — DISMISSED** [review]
Change Procfile too or not?

File: mergify_engine/web/asgi.py [hunk 5]
```diff
@@ -22,4 +22,5 @@
 if config.SENTRY_URL:
     application = SentryAsgiMiddleware(application)  # type:ignore[assignment]
 
+ddtrace.config.service = "web"
```
Comment (jd) [inline mergify_engine/web/asgi.py:25]:
```suggestion
ddtrace.config.service = "engine-web"
```

**Review by jd — CHANGES_REQUESTED** [review]

File: mergify_engine/worker.py [hunk 6]
```diff
@@ -1036,6 +1050,7 @@ async def run_forever() -> None:
 
 
 def main() -> None:
+    ddtrace.config.service = "worker"
```
Comment (jd) [inline mergify_engine/worker.py:1053]:
```suggestion
    ddtrace.config.service = "engine-worker"
```

**Review by jd — CHANGES_REQUESTED** [review]

File: mergify_engine/web/asgi.py (hunk 3 above)
Comment (sileht) [reply mergify_engine/web/asgi.py]: This starts to be messy.
For agent and process plugin we use:
  - appname:$HEROKU_APP_NAME
  - service:web/worker

And for logging just :
      service: mergify-engine

Comment (jd) [reply mergify_engine/web/asgi.py]: Well `appname` is Heroku specific and not use across Datadog's product, so we can ignore it basically.

Our current service naming in Datadog is kinda wrong and is not going to give a good experience in APM.

Comment (sileht) [reply mergify_engine/web/asgi.py]: gotcha

**Review by jd — APPROVED** [review]

**Pull request merged**